src/components/Details/Modal.js

```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import DatePicker from 'react-datepicker';
+import reservedItems from '../../api/reservedItems';
+import 'react-datepicker/dist/react-datepicker.css';
+import './Details.css';
+import { controlModal } from '../../redux/items/ItemDetails';
+
+const Modal = () => {
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  const [loginResponse, setLoginResponse] = useState('');
+  const { id, name } = useParams();
+  const dispatch = useDispatch();
+
+  const user = useSelector((state) => state.usersReducer.user);
+  const reservedDates = useSelector((state) => state.itemsDetailsReducer.dates);
+  const tope = [];
+  reservedDates.forEach((date) => {
+    const startDate = date.start_date.split('-');
+    const endDate = date.end_date.split('-');
+    let startMonth = parseInt(startDate[1], 10) - 1;
+    startMonth = String(startMonth);
+    let endMonth = parseInt(endDate[1], 10) - 1;
+    endMonth = String(endMonth);
+    tope.push({
+      start: new Date(startDate[0],
+        startMonth,
+        startDate[2]),
+      end: new Date(endDate[0],
+        endMonth, endDate[2]),
+    });
+  });
+
+  const reserveSubmit = async (e) => {
+    e.preventDefault();
+    const response = await reservedItems({
+      start_date: startDate,
+      end_date: endDate,
+      user_id: user.id,
+      id: parseInt(id, 10),
+    });
+    if (!response.error) {
+      setLoginResponse('Succesfully Reserved');
+    } else {
+      setLoginResponse(response.error);
+    }
```

Comment:
**[REQUIRED]** - When a user makes a reservation for an item, please ensure that he is redirected to the reservations page. This will make for a better User experience. Also, Please fix this for the reservation form page.

Reply:
Thank you for the suggestion! Fixed it.